Source/Task/WaitTimer_stl.cpp

```diff
@@ -186,27 +188,61 @@ namespace OS
         return e;
     }
 
+    WaitTimerImpl::~WaitTimerImpl()
+    {
+        std::lock_guard<std::mutex> lock{ g_timerQueueMutex };
+
+        // If we are the last one referencing the global timer the
+        // shared use count will be two (us + the global). If it is,
+        // clear out the global. We let our own reference reset
+        // as the class destructs. This puts it outside the mutex
+        // lock, which we want since there is some shutdown cost
+        // associated with shutting the timer down.
+
+        if (g_timerQueue.use_count() == 2)
+        {
+            g_timerQueue.reset();
+        }
+    }
+
     HRESULT WaitTimerImpl::Initialize(_In_opt_ void* context, _In_ WaitTimerCallback* callback)
     {
         m_context = context;
         m_callback = callback;
 
-        if (!g_timerQueue.LazyInit())
+        std::lock_guard<std::mutex> lock{ g_timerQueueMutex };
+
+        if (g_timerQueue == nullptr)
         {
-            return E_FAIL;
+            try
+            {
+                auto queue = std::make_shared<TimerQueue>();
```

Comment:
change to:
                auto queue = http_allocate_shared<TimerQueue>();
                
this will use the mem hooked alloc.  see global.cpp for example usage